agent/eventhandler/handler.go

```diff
@@ -30,27 +30,20 @@ func HandleEngineEvents(taskEngine engine.TaskEngine, client api.ECSClient, save
 	statesaver = saver
 
 	for {
-		taskEvents, containerEvents := taskEngine.TaskEvents()
+		stateChangeEvents := taskEngine.StateChangeEvents()
```

Comment:
I didn't see why this needs to be called inside every loop, couldn't this be moved outside the loop?